src/index.spec.ts

```diff
@@ -179,6 +186,217 @@ describe('contracts', () => {
   });
 });
 
+describe('check errors', () => {
+  it('tuple type', () => {
+    try {
+      const Check = Tuple(Number, String, Boolean);
+      Check.check([false, '0', true]);
+      fail("tuple passed, eventhough should've failed");
+    } catch (e) {
+      expect(e.message).toBe('Expected a number, but was boolean');
+      expect(e.key).toBe('[0]');
+    }
+  });
+
+  it('tuple length', () => {
+    try {
+      const Check = Tuple(Number, String, Boolean);
+      Check.check([0, '0']);
+      fail("tuple passed, eventhough should've failed");
+    } catch (e) {
+      expect(e.message).toBe('Expected an array of 3, but was 2');
+      expect(e.key).toBe(undefined);
+    }
+  });
+
+  it('tuple nested', () => {
+    try {
+      const Nested = Record({ name: String });
+      const Check = Tuple(Number, Nested);
+      Check.check([0, { name: 0 }]);
+      fail("tuple passed eventough should've failed");
+    } catch (e) {
+      expect(e.message).toBe('Expected a string, but was number');
+      expect(e.key).toBe('[1].name');
+    }
+  });
+
+  it('array', () => {
+    try {
+      const Check = Array(Number);
+      Check.check([0, 2, 'test']);
+      fail("array passed eventough should've failed");
+    } catch (e) {
+      expect(e.message).toBe('Expected a number, but was string');
+      expect(e.key).toBe('[2]');
+    }
+  });
+
+  it('array nested', () => {
+    try {
+      const Nested = Record({ name: String });
+      const Check = Array(Nested);
+      Check.check([{ name: 'Foo' }, { name: false }]);
+      fail("array passed eventough should've failed");
+    } catch (e) {
+      expect(e.message).toBe('Expected a string, but was boolean');
+      expect(e.key).toBe('[1].name');
+    }
+  });
+
+  it('array null', () => {
+    try {
+      const Nested = Record({ name: String });
+      const Check = Array(Nested);
+      Check.check([{ name: 'Foo' }, null]);
+      fail("array passed eventough should've failed");
+    } catch (e) {
+      expect(e.message).toBe('Expected { name: string; }, but was null');
+      expect(e.key).toBe('[1]');
+    }
+  });
+
+  it('dictionary', () => {
+    try {
+      const Check = Dictionary(String);
+      Check.check(null);
+      fail("dictionary passed eventough should've failed");
+    } catch (e) {
+      expect(e.message).toBe('Expected a dictionary [string: string], but was null');
+      expect(e.key).toBe(undefined);
+    }
+  });
+
+  it('dictionary invalid type', () => {
+    try {
+      const Value = Record({ name: String });
+      const Check = Dictionary(Value);
+      Check.check(undefined);
+      fail("dictionary passed eventough should've failed");
+    } catch (e) {
+      expect(e.message).toBe(
+        'Expected a dictionary [string: { name: string; }], but was undefined',
```

Comment:
`[string: { name: string; }]` doesn't parse as TypeScript. Better to just serialize the type:

>Expected `{ name: string }` but was `undefined`.